### ISSUE TITLE
`listchars` eol displayed by default

### ISSUE THREAD

**Issue by deivid-rodriguez** [neovim/neovim#4048]
```shell
$ nvim --version
NVIM 0.1.2-dev
Build type: RelWithDebInfo
Compilation: /usr/bin/x86_64-linux-gnu-gcc -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -I/build/neovim-41ccc3/neovim-0.1.1ubuntu1+git201601181513+2173+16~ubuntu15.10.1/build/config -I/build/neovim-41ccc3/neovim-0.1.1ubuntu1+git201601181513+2173+16~ubuntu15.10.1/src -I/build/neovim-41ccc3/neovim-0.1.1ubuntu1+git201601181513+2173+16~ubuntu15.10.1/.deps/usr/include -I/build/neovim-41ccc3/neovim-0.1.1ubuntu1+git201601181513+2173+16~ubuntu15.10.1/.deps/usr/include -I/usr/include/luajit-2.0 -I/build/neovim-41ccc3/neovim-0.1.1ubuntu1+git201601181513+2173+16~ubuntu15.10.1/.deps/usr/include -I/build/neovim-41ccc3/neovim-0.1.1ubuntu1+git201601181513+2173+16~ubuntu15.10.1/.deps/usr/include -I/build/neovim-41ccc3/neovim-0.1.1ubuntu1+git201601181513+2173+16~ubuntu15.10.1/.deps/usr/include -I/build/neovim-41ccc3/neovim-0.1.1ubuntu1+git201601181513+2173+16~ubuntu15.10.1/.deps/usr/include -I/usr/include -I/build/neovim-41ccc3/neovim-0.1.1ubuntu1+git201601181513+2173+16~ubuntu15.10.1/build/src/nvim/auto -I/build/neovim-41ccc3/neovim-0.1.1ubuntu1+git201601181513+2173+16~ubuntu15.10.1/build/include
Compilado por buildd@lgw01-22

Optional features included (+) or not (-): +acl   +iconv    +jemalloc 
For differences from Vim, see :help vim-differences

   archivo «vimrc» del sistema: "$VIM/sysinit.vim"
  predefinido para $VIM: "/usr/share/nvim"
```

If I do

```shell
$ nvim -u NORC -c 'set list' -c 'hi NonText ctermbg=253 ctermfg=16'
```

the new line character is highlighted. 

`help 'listchars'` tells me that the default setting omits `eol` and that when `eol` is omitted, there is no extra character at the end of the line, so my expectation would be that nothing is highlighted.

I think this is a recent regression, since I'm tracking master and I started seeing this only recently.

Please let me know if I'm misunderstanding something and this is not a bug, or if the cause of this is not obvious for you and you need my to git bisect and find the commit that changed the behaviour. 

Also, I thought #4041 could be related, but not sure it's a duplicate.

**Comment by justinmk** [comment]
I think this was already reported at https://github.com/neovim/neovim/issues/4040 , let me know if not. Will be fixed by 7.4.729

**Comment by deivid-rodriguez** [comment]
Yes, @justinmk, definitely the same issue. I'll watch that one, thanks a lot!
